Add empty offerings troubleshooting wizard (#654)
RevenueCat/docs: docs/offerings/troubleshooting.mdx

A common issue after configuring your products in the RevenueCat dashboard is that the products aren't able to be retrieved from the SDK. Typically, this is due to a configuration issue.

<YouTubeEmbed videoId="HbF-5tpaBzc" title="Troubleshooting Offerings" />

Here's how it works under the hood:

- Products are configured in the RevenueCat dashboard.
- The SDK retrieves the product identifiers from the RevenueCat API.
- The SDK uses that response to retrieve the products from the store on the device.

Since the actual products that your users purchase can only be retrieved from the stores directly on the device, any inconsistencies in the configuration between RevenueCat and the stores will prevent the products from being retrieved for purchase.

<TroubleshootingWizard
  title="Troubleshooting Wizard"
  subtitle="Follow these steps to verify your product configuration"
  platforms={[
    { id: 'ios', name: 'iOS / Apple Platforms', icon: '' },
    { id: 'android', name: 'Android / Google Play', icon: '🤖' },
    { id: 'amazon', name: 'Android / Amazon Appstore', icon: '📦' }
  ]}
  steps={[
    {
      title: "Check App Store Connect Agreements",

>       description: "Verify that all agreements (including the Paid Applications Agreement) have been signed and banking details have been completed in App Store Connect.",

Author: ```suggestion
      description: "Verify that all agreements (including the Paid Applications Agreement) in App Store Connect > Business have been signed and banking details have been completed in App Store Connect. This can take up to 24 hours to propagate after completing, including if your developer program expires and is then renewed.",
```

      caption: "Note: The Paid Applications Agreement may have expired - please check its status.",
      platforms: ['ios']
    },
    {
      title: "Check Google Play Console Agreements",
      description: "Verify that all agreements have been signed and banking details have been completed in Google Play Console.",
      platforms: ['android']
    },
    {
      title: "Published on a Closed Track",
      description: "Ensure your app is published on a closed track in Google Play Console. Google Play requires all apps to be published on a closed track before products can be fetched, even when just testing.",
      platforms: ['android']
    },
    {
      title: "Verify Product Status",

>       description: "Ensure the products are in the 'Ready to Submit' or 'Approved' state in App Store Connect.",

Reviewer: Also worth adding this quote from Apple docs:
> If you’re submitting your first in-app purchase or adding a new [in-app purchase type](https://developer.apple.com/help/app-store-connect/reference/in-app-purchase-types) to your app, you’ll need to submit it with a new version of your app.

>       description: "Ensure the products are in the 'Ready to Submit' or 'Approved' state in App Store Connect.",

Author: ```suggestion
      description: "Ensure the products are in the 'Ready to Submit' or 'Approved' state in App Store Connect. If you're submitting your first in-app purchase or adding a new in-app purchase type to your app, you'll need to submit it with a new version of your app.",
```

      platforms: ['ios']
    },
    {
      title: "Product Identifier Match",
      description: "Confirm the product identifier in the RevenueCat dashboard exactly matches the product identifier in your store.",
      platforms: ['ios', 'android', 'amazon']
    },
    {

>       title: "Physical Device Testing",

Author: done!

>       description: "Confirm the product identifier in the RevenueCat dashboard exactly matches the product identifier in your store.",

Author: ```suggestion
      description: "Confirm the product identifiers in the RevenueCat dashboard exactly matches the product identifiers in your store.",
```

>       description: "Ensure you're testing on a physical device and not a simulator.",

Author: ```suggestion
      description: "Ensure you're testing on a physical device and not a simulator. Additionally, ensure you're logged into a sandbox Apple Account on the device (not a production Apple Account).",
```

      caption: "Simulators do not have access to the live App Store API, and will only work with StoreKit Configuration files.",
      platforms: ['ios']
    },
    {
      title: "Physical Device Testing",

>       description: "Ensure you're testing on a physical device and not a simulator.",

Author: ```suggestion
      description: "We strongly recommend testing on a physical device and not a simulator.",
```

      caption: "Simulators do not have access to the live store APIs.",
      platforms: ['android', 'amazon']
    },
    {
      title: "Bundle ID Validation",
      description: "Verify the Bundle ID in the RevenueCat dashboard matches the Bundle ID in App Store Connect and your Xcode project exactly.",
      caption: "This is case-sensitive. If you're using multiple environments with different Bundle IDs, be sure to check the Bundle ID in each environment.",
      platforms: ['ios']
    },
    {
      title: "Package Name Validation",
      description: "Verify the package name in the RevenueCat dashboard matches the package name in Google Play Console and your Android project exactly.",
      caption: "This is case-sensitive. If you're using multiple environments with different package names, be sure to check the package name in each environment.",
      platforms: ['android', 'amazon']
    },
    {
      title: "API Key Platform Check",

>       description: "Confirm you're using the correct platform-specific API key. The RevenueCat SDK uses different API keys for each platform.",

Author: ```suggestion
      description: "Configure the RevenueCat SDK with the correct app-specific API key. The RevenueCat SDK uses different API keys for each platform.",
```

<TroubleshootingWizard
  title="Troubleshooting Guide"
  subtitle="Follow these steps to verify your product configuration"
  platforms={[
    { id: 'ios', name: 'iOS / Apple Platforms', icon: '' },
    { id: 'android', name: 'Android / Google Play', icon: '🤖' },
    { id: 'amazon', name: 'Android / Amazon Appstore', icon: '📦' }
  ]}
  steps={[
    {
      title: "Check App Store Connect Agreements",
      description: "Verify that all agreements (including the Paid Applications Agreement) in App Store Connect > Business have been signed and banking details have been completed in App Store Connect. This can take up to 24 hours to propagate after completing, including if your developer program expires and is then renewed.",
      caption: "Note: The Paid Applications Agreement may have expired - please check its status.",
      platforms: ['ios']
    },
    {
      title: "Check Google Play Console Agreements",
      description: "Verify that all agreements have been signed and banking details have been completed in Google Play Console.",
      platforms: ['android']
    },
    {
      title: "Published on a Closed Track",

>       description: "Ensure your app is published on a closed track in Google Play Console. Google Play requires all apps to be published on a closed track before products can be fetched, even when just testing.",

Reviewer 2: ```suggestion
      description: "Ensure your app is published on a [closed track](https://www.revenuecat.com/docs/test-and-launch/sandbox/google-play-store#create-a-closed-track-and-add-a-tester-to-it) in Google Play Console. Google Play requires all apps to be published on a closed track before products can be fetched, even when just testing.",
```

      platforms: ['android']
    },
    {
      title: "Licensed Testers",
      description: "Add a licensed tester in Google Play Console > Settings > License testing. Add here the account you are using in your real device (the account you are logged in with).",
      caption: "Read our guide here: <a href='https://www.revenuecat.com/docs/test-and-launch/sandbox/google-play-store#create-a-test-user-and-configure-licensing-testing' target='_blank' rel='noopener noreferrer'>Create a Test User and Configure Licensed Testing</a>",
      platforms: ['android']
    },
    {
      title: "Verify Product Status",

>       description: "Ensure the products are in the 'Ready to Submit' or 'Approved' state in App Store Connect. If you're submitting your first in-app purchase or adding a new in-app purchase type to your app, you'll need to submit it with a new version of your app.",

Reviewer 2: ```suggestion
      description: "Ensure the products are in the 'Ready to Submit' or 'Approved' state in App Store Connect. If you're submitting your first in-app purchase or adding a new in-app purchase type to your app, you'll need to submit it for review with a new version of your app. _Please ensure you **do not** separately submit your products for review without the app itself._",
```

      platforms: ['ios']
    },
    {
      title: "Product Identifier Match",

>       description: "Confirm the product identifiers in the RevenueCat dashboard exactly matches the product identifiers in your store.",

Reviewer 2: ```suggestion
      description: "Confirm the product identifiers in the RevenueCat dashboard exactly match the product identifiers in your store.",
```

      platforms: ['ios', 'android', 'amazon']
    },
    {
      title: "Physical Device Testing",

>       description: "We strongly recommend testing on a physical device and not a simulator. Additionally, ensure you're logged into a sandbox Apple Account on the device (not a production Apple Account).",

Reviewer 2: ```suggestion
      description: "We strongly recommend testing on a physical device and not a simulator. Additionally, ensure you're logged into a [sandbox](https://www.revenuecat.com/docs/test-and-launch/sandbox/apple-app-store) Apple Account on the device (not a production Apple Account).",
```

      caption: "Simulators do not have access to the live App Store API, and will only work with StoreKit Configuration files.",
      platforms: ['ios']
    },
    {
      title: "Physical Device Testing",
      description: "We strongly recommend testing on a physical device and not a simulator.",
      caption: "Simulators do not have access to the live store APIs.",
      platforms: ['android', 'amazon']
    },
    {
      title: "Bundle ID Validation",
      description: "Verify the Bundle ID in the RevenueCat dashboard matches the Bundle ID in App Store Connect and your Xcode project exactly.",

>       caption: "This is case-sensitive. If you're using multiple environments with different Bundle IDs, be sure to check the Bundle ID in each environment.",

Reviewer 2: ```suggestion
      caption: "This is case-sensitive. If you're using [multiple environments](https://www.revenuecat.com/docs/guides/environment-strategies) with different Bundle IDs, be sure to check the Bundle ID in each environment.",
```

      platforms: ['ios']
    },
    {
      title: "Package Name Validation",
      description: "Verify the package name in the RevenueCat dashboard matches the package name in Google Play Console and your Android project exactly.",

>       caption: "This is case-sensitive. If you're using multiple environments with different package names, be sure to check the package name in each environment.",

Reviewer 2: ```suggestion
      caption: "This is case-sensitive. If you're using [multiple environments](https://www.revenuecat.com/docs/guides/environment-strategies) with different package names, be sure to check the package name in each environment.",
```

      platforms: ['android', 'amazon']
    },
    {
      title: "API Key Platform Check",

>       description: "Configure the RevenueCat SDK with the correct app-specific API key. The RevenueCat SDK uses different API keys for each platform.",

Reviewer 2: ```suggestion
      description: "Configure the RevenueCat SDK with the correct app-specific [API key](https://www.revenuecat.com/docs/projects/authentication). The RevenueCat SDK uses different API keys for each platform.",
```

      caption: "For example, if you use a Google key for iOS, the wrong product configuration will be retrieved, and products will not be able to be fetched.",
      platforms: ['ios', 'android', 'amazon']
    },
    {
      title: "StoreKit Configuration",
      description: "Confirm your Xcode scheme does not have a StoreKit Configuration file configured if attempting to test with the live App Store API.",
      platforms: ['ios']
    },
    {
      title: "Recent App Release",

>       description: "If you've recently released your app, please note that products may not be available for up to 24 hours after release.",

Reviewer 2: ```suggestion
      description: "If you've recently released your app with new in-app products, please note that these products may not be available for up to 24 hours after release.",
```
